lib/config/presets/index.spec.ts

```diff
@@ -288,6 +288,30 @@ describe('config/presets/index', () => {
       expect(local.getPreset.mock.calls[0][0].baseConfig).toBeDefined();
       expect(res).toMatchSnapshot();
     });
+
+    it('resolves self-hosted transitive presets without baseConfig', async () => {
+      config.platform = 'gitlab';
+      config.endpoint = 'https://dummy.example.com/api/v4';
+      config.extends = ['local>username/preset-repo'];
+      local.getPreset = jest
+        .fn()
+        .mockImplementationOnce((repo, presetName, baseConfig) =>
+          Promise.resolve({
+            extends: ['local>username/preset-repo//subpreset'],
+          })
+        )
+        .mockImplementation((repo, presetName, baseConfig) =>
+          Promise.resolve({ labels: ['self-hosted resolved'] })
+        );
```

Comment:
```suggestion
      local.getPreset
        .mockResolvedValueOnce(
          {
            extends: ['local>username/preset-repo//subpreset'],
          }
        )
        .mockResolvedValueOnce({ labels: ['self-hosted resolved'] });
```

simplify

Reply:
I refactored as of your recommendations. However, doing so revealed that the mocking that is used in the previous tests has side effects to my test. So I needed to specifically call `mockClear`. As the previous test had the same issue/pattern I opted to refactor this as well (although it is unrelated to my fix).